fix(EuiCodeBlock):  fixes re-renders causing blank lines to appear when using isVirtualized if scrolled

sravanth-space opened the pull request:
## Summary
- Fixes blank rendering in virtualized CodeBlock when scrolled (#9034)
- Moves accessibility label from VirtualizedOuterElement to VirtualizedInnerElement
- Preserves accessibility while preventing react-window calculation issues

Partially reverts the implementation approach from PR #8887 while maintaining the accessibility improvements.

## Why are we making this change?
Problem:
After PR #8887, the accessibility label was injected inside the \<pre\> element managed by react-window's virtualization as part of the VirtualizedOuterElement. This caused react-window to miscalculate scroll offsets, resulting in blank lines appearing when:
- User scrolls down in a virtualized code block
- Content is updated (e.g., build logs appending new lines)
- The component re-renders

Root Cause:
react-window's FixedSizeList requires the outer element to be "pure" - any injected content disrupts its internal measurements and scroll position calculations.

Solution:
- Move the accessibility label from `VirtualizedOuterElement` to `VirtualizedInnerElement` to ensure a stable container for the virtualized calculations
  - the label remains a child of the `pre` element and hence is read as label when the code element is focused
  - This ensures the outer virtualized container stays clean for proper scroll calculations while maintaining accessibility

## Screenshots <a href="#user-content-screenshots" id="screenshots">#</a>

| before | after |
|---|---|
| <img width="1412" height="868" alt="image" src="https://github.com/user-attachments/assets/8f41e6bf-a1d0-4f02-962b-05981283d10d" /> | <img width="1412" height="868" alt="image" src="https://github.com/user-attachments/assets/3ab693ab-b6ed-41fe-af7c-66b06484729b" /> |


<!--
If this change includes changes to UI, it is important to include screenshots or gif. This helps our users understand what changed when reviewing our changelogs.
-->
Its best to use the sandbox mentioned in the issue https://codesandbox.io/p/devbox/euicodeblock-virtualized-within-euiflyout-height-100-forked-6j2tlc?workspaceId=ws_PBRiDLztB2gZ9hEEoiqpdS and change it with this pr patch.

## Impact to users

<!--
How will this change impact EUI users? If it's a breaking change, what will they need to do to handle this change when upgrading? Take a moment to look at usage in Kibana and consider how many usages this will impact and note it here.

Even if it is not a breaking change, how significant is the visual change? Is it a large enough visual change that we would want them advise them to test it?
-->
Fixes a bug where virtualized code blocks would show blank lines when scrolled or when content updates. No breaking changes.

🟢 There are no updates required on consumers side.

## QA

Remove or strikethrough items that do not apply to your PR.

### General checklist

- Browser QA
    - [x] Checked for **accessibility** including keyboard-only and screenreader modes
- Code quality checklist
    - [x] ~~Added or updated **[jest](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/unit-testing.md) and [cypress](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/cypress-testing.md) tests**~~
    - [x] Updated **[visual regression tests](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/testing/visual-regression-testing.md)**
- Release checklist
    - [x] A **[changelog](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting/changelogs.md)** entry exists and is marked appropriately

mgadewoll commented:
@sravanth-space Please be so kind to also update the PR description according to the final updates 🙏

sravanth-space commented:
> @sravanth-space I added a VRT-only story as regression test for this bug, I hope you don't mind me pushing it directly.

Thanks. I have updated the PR description and for the jest and cypress tests I haven't added/updated please let me know if I need to add/updated any.

mgadewoll reviewed (approved):
🟢 The changes are LGTM and the fix works correctly. Awesome work! 
Thanks for contributing! 🎉

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/5212)
* Commit: 912a330cd11c885d52efc99af83633a1b441ff03

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/5212","number":5212,"commit":"912a330cd11c885d52efc99af83633a1b441ff03"}],"number":5212}
buildkite-pr-comment-->
